### PR TITLE
WIP: Custom serializer field: DictRelatedField

### PR DESCRIPTION
**Pull request by JureZmrzlikar**

### CODE REVIEW

File: resolwe/flow/serializers/fields.py [hunk 1]
```diff
@@ -11,63 +11,65 @@
...
+        else:
+            self.fail('no_data', name=self.field_name)
+
         try:
             return get_objects_for_user(
                 user,
                 permission,
-                queryset.filter(**{self.slug_field: data}),
+                queryset.filter(**kwargs),
             ).latest()
```
Comment (mstajdohar) [inline resolwe/flow/serializers/fields.py]: Is this OK? I know we set `get_latest_by` in the base model. What if one of the models overrides that, or if we use this field in other use cases at some point. Do we want to be more explicit here and pass `version` to `latest()`?

**Comment by mstajdohar** [comment]
Don't forget about the change log—very important in this case.

File: resolwe/flow/serializers/fields.py [hunk 2]
```diff
@@ -11,63 +11,65 @@
 from .descriptor import DescriptorSchemaSerializer
 
 
-class ResolweSlugRelatedField(RelatedField):
+class DictRelatedField(RelatedField):
```
Comment (mstajdohar) [inline resolwe/flow/serializers/fields.py]: Should we call this `ResolweRelatedField`? That would be a dict field that can contain either id or slug, and can be hydrated or not.

**Comment by JureZmrzlikar** [comment]
Closed in favor of [#631](https://github.com/genialis/resolwe/pull/631)

**Pull request closed without merging**